Subject: [PATCH] fix cdrom autoclose
X-BK-ChangeSetKey: stsp@aknet.ru[torvalds]|ChangeSet|20050108214119|41527
Date:	Sat, 08 Jan 2005 23:14:51 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2272, 2005/01/08 13:41:19-08:00, stsp@aknet.ru

	[PATCH] fix cdrom autoclose
	
	The attached patch fixes the CD-ROM autoclose.  It is broken in recent
	kernels for CD-ROMs that do not properly report that the tray is opened.
	Now on such a drives the kernel will do one close attempt and check for the
	disc again.  This is how it used to work in the past.
	
	Signed-off-by: Stas Sergeev <stsp@aknet.ru>
	Acked-by: Alexander Kern <alex.kern@gmx.de>
	Acked-by: Jens Axboe <axboe@suse.de>
	Signed-off-by: Andrew Morton <akpm@osdl.org>
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 cdrom/cdrom.c |    2 ++
 ide/ide-cd.c  |    4 ++--
 2 files changed, 4 insertions(+), 2 deletions(-)


diff -Nru a/drivers/cdrom/cdrom.c b/drivers/cdrom/cdrom.c
--- a/drivers/cdrom/cdrom.c	2005-01-08 15:09:33 -08:00
+++ b/drivers/cdrom/cdrom.c	2005-01-08 15:09:33 -08:00
@@ -1076,6 +1076,8 @@
 			}
 			cdinfo(CD_OPEN, "the tray is now closed.\n"); 
 		}
+		/* the door should be closed now, check for the disc */
+		ret = cdo->drive_status(cdi, CDSL_CURRENT);
 		if (ret!=CDS_DISC_OK) {
 			ret = -ENOMEDIUM;
 			goto clean_up_and_return;
diff -Nru a/drivers/ide/ide-cd.c b/drivers/ide/ide-cd.c
--- a/drivers/ide/ide-cd.c	2005-01-08 15:09:33 -08:00
+++ b/drivers/ide/ide-cd.c	2005-01-08 15:09:33 -08:00
@@ -2688,9 +2688,9 @@
 	 */
 	if (sense.sense_key == NOT_READY) {
 		if (sense.asc == 0x3a) {
-			if (sense.ascq == 0 || sense.ascq == 1)
+			if (sense.ascq == 1)
 				return CDS_NO_DISC;
-			else if (sense.ascq == 2)
+			else if (sense.ascq == 0 || sense.ascq == 2)
 				return CDS_TRAY_OPEN;
 		}
 	}
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

